app/models/walletkey.js

```diff
@@ -0,0 +1,24 @@
+const mongoose = require('mongoose');
+const _ = require('lodash');
+
+const walletKeySchema = new mongoose.Schema({
+  masterKey: { type: mongoose.Types.ObjectId },
+  path: { type: mongoose.Types.String },
+  xpub: { type: mongoose.Types.String },
+  userEmail: { type: mongoose.Types.String },
+  notificationUrl: { type: mongoose.Types.String },
+  verificationInfo: { type: mongoose.Types.String },
+  custom: {}
+});
+
+walletKeySchema.methods = {
+  toJSON: function() {
+    return _.pick(this, ['masterKey', 'path', 'xpub', 'userEmail', 'notificationUrl', 'verificationUrl', 'custom']);
+  }
+};
+
+walletKeySchema.index({ masterKey: 1, path: 1 }, { unique: true });
```

Comment:
Is it necessary to have/index by masterKey&path? If you had the masterKey xpub and the path, couldn't we just generate the walletKey xpub from that and just search by that?

Reply:
will remove and add utility to admin tool for deriving individual xpubs, good idea.